File: backend/app/core/event_types.py
# docs/event-taxonomy.md에 정의된 이벤트 이름만 허용한다 (CLAUDE.md 규칙 7: 임의 변경 금지).
ALLOWED_EVENT_TYPES: frozenset[str] = frozenset(
    {
        "ingredient_search",
        "ingredient_added",
        "ingredient_selected",
        "freshness_selected",
        "recommend_request",
        "recommendation_impression",
        "recipe_click",
        "recipe_detail_view",
        "recipe_save",
        "recipe_start",
        "recipe_complete",
    }
)

# GET /api/v1/recipes/recent 가 "최근 본 레시피"로 취급하는 이벤트.
RECENT_RECIPE_VIEW_EVENT_TYPES: frozenset[str] = frozenset({"recipe_detail_view", "recipe_click"})

File: frontend/src/features/analytics/eventTypes.ts
// docs/event-taxonomy.md에 정의된 11개 이벤트 이름. 임의로 바꾸지 않는다 (CLAUDE.md 규칙 7).
export const EVENT_TYPES = [
  'ingredient_search',
  'ingredient_added',
  'ingredient_selected',
  'freshness_selected',
  'recommend_request',
  'recommendation_impression',
  'recipe_click',
  'recipe_detail_view',
  'recipe_save',
  'recipe_start',
  'recipe_complete',
] as const

export type EventType = (typeof EVENT_TYPES)[number]
